In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
df = pd.read_excel("cleaned_dataset.xlsx")

In [3]:
df.head()

,Order_ID,Order_Date,Customer_ID,Customer_Name,Age,Gender,City,Product,Category,Quantity,Unit_Price,Total_Sales
0,ORD100002,2025-02-25,CUST5529,Customer_227,30,Female,Bengaluru,Rice,Grocery,7,2829.77,19808.39
1,ORD100003,2025-10-14,CUST3127,Customer_182,63,Male,Bengaluru,Book,Education,5,27906.16,139530.80
2,ORD100004,2025-05-13,CUST8887,Customer_487,62,Female,Bengaluru,Book,Education,8,37491.06,299928.48
3,ORD100005,2025-12-02,CUST2515,Customer_470,65,Female,Kolkata,Mobile,Electronics,9,28541.36,256872.24
4,ORD100006,2025-11-20,CUST4796,Customer_380,44,Male,Bengaluru,Rice,Grocery,10,14036.59,140365.90


# Statistical Analysis

In [4]:
df["Total_Sales"].describe()

count      1000.000000
mean     139399.439650
std      114100.051546
min         437.340000
25%       47066.632500
50%      108594.025000
75%      203722.882500
max      493677.500000
Name: Total_Sales, dtype: float64

In [5]:
df["Total_Sales"].mean()

np.float64(139399.43965000001)

In [6]:
df["Total_Sales"].median()

108594.025

In [7]:
df["Total_Sales"].std()

114100.05154617467

In [8]:
df["Gender"].value_counts()

Gender
Male      511
Female    489
Name: count, dtype: int64

In [10]:
male_sales = df[df["Gender"] == "Male"]["Total_Sales"]
female_sales = df[df["Gender"] == "Female"]["Total_Sales"]

In [11]:
from scipy.stats import ttest_ind

t_stat, p_value = ttest_ind(male_sales, female_sales)

print("T-Statistic:", t_stat)
print("P-Value:", p_value)

T-Statistic: 0.6820125406168254
P-Value: 0.4953892684482486


# Hypothesis Testing

## Objective
To determine whether there is a significant difference in Total Sales between Male and Female customers.

## Null Hypothesis (H₀)
There is no significant difference in average Total Sales between Male and Female customers.

## Alternative Hypothesis (H₁)
There is a significant difference in average Total Sales between Male and Female customers.

## Results
T-Statistic: 0.682
P-Value: 0.495

## Conclusion
Since the p-value (0.495) is greater than 0.05, we fail to reject the null hypothesis.

Therefore, there is no statistically significant difference in Total Sales between Male and Female customers.

# Data Storytelling

## Business Objective
Analyze customer sales data to identify trends and insights that can support business decision-making.

## Key Findings
- Total records analyzed: 1000
- Average Total Sales: 139,399.44
- Median Total Sales: 108,594.03
- Sales values show high variability.
- Male customers: 511
- Female customers: 489
- No significant sales difference between genders.

## Business Impact
Customer purchasing behavior appears similar across genders. Business strategies should focus on product categories, customer segments, and purchasing patterns rather than gender-based targeting.

## Recommendations
- Focus marketing efforts on high-performing product categories.
- Improve promotions for low-performing categories.
- Continue monitoring customer purchasing trends.
- Use data-driven strategies for customer engagement.

## Conclusion
The analysis provides valuable insights into sales performance and customer behavior. Statistical testing confirms that gender does not significantly influence Total Sales in this dataset.